[samples_index] Fix search bug when using filters

cazci opened the pull request:
Fixes #392 

Currently, the search results are inaccurate when type filters are used. I've fixed this by adding an additional type check before the current checks. This loop will check whether a type parameter is present in the search query, and return false if the current self type mismatches the query type. Hope this helps :)

I've tested this against a set of custom queries, and the following is a screenshot when tested against the query mentioned in the issue.
`?search=ios&type=cookbook`

![sample_index](https://user-images.githubusercontent.com/32796120/99288395-89934180-2861-11eb-8c6c-f35179c8d0ef.png)

johnpryan reviewed (approved):
Looks good, want to add a test to [this file](https://github.com/flutter/samples/blob/master/web/samples_index/test/samples_index_test.dart#L99)?

cazci commented:
Sure, I'll add

cazci commented:
I've added three tests and squashed the commits. Test output is as follows, 

![tests](https://user-images.githubusercontent.com/32796120/99425969-8fecf080-2929-11eb-907b-8da19339bb7c.png)

johnpryan reviewed (approved):
looks good to me, thanks!
